src/gluonts/distribution/distribution.py

```diff
@@ -253,6 +253,19 @@ def quantile(self, level: Tensor) -> Tensor:
         """
         raise NotImplementedError()
 
+    def slice_axis(
+        self, axis: int, begin: int, end: Optional[int]
+    ) -> "Distribution":
+        """
+        Construct a new distribution by slicing all constructor arguments
+        as specified by the provided bounds. Relies on ``mx.nd.slice_axis``.
+        """
+        sliced_distr = self.__class__(
+            *[a.slice_axis(axis, begin, end) for a in self.args]
+        )
+        assert isinstance(sliced_distr, type(self))
+        return sliced_distr
```

Comment:
I could use `type(self)` instead of `self.__class__` and return the object directly. However, that way mypy complains that "too many constructor arguments are passed", so it probably infers that `type(self) == Distribution` instead of the most specific class. This way instead, I get to fool mypy (a-ha!).

Reply:
yes